Sources/HaCWebsiteLib/NewWorkshopManager.swift

```diff
@@ -17,15 +17,40 @@ public struct NewWorkshopManager {
     attributes: .concurrent
   )
 
+  private static var pollSource: DispatchSourceTimer! = nil
+  private static var currentlyUpdating = false
+
   /// The list of Workshops as derived from the workshop repos last time they were processed. Keys are workshop repo names
   public private(set) static var workshops: [String: NewWorkshop] = [:]
 
   /// All the workshop repos that we know about. Keys are workshop repo names
   private static var workshopRepoUtils: [String: GitUtil] = [:]
 
   public static func update() {
-    updateRepoList()
-    updateWorkshopsData()
+    print("Updating workshops...")
+    // Prevent multiple updates at once (might happen if an update takes longer than the poll interval)
+    if !currentlyUpdating {
```

Comment:
Are there any concurrency issues involved here?

Reply:
The only race is if a user triggers an update at the same time as an automatic update. This is mostly an optimisation trying to protect from queuing updates that we don't need rather than any real safety concern. This is unlikely to ever cause a problem but I can make this more robust and solve the polling abstraction thing below by creating a PollUtil that handles all this for us.

Reply:
We can solve it later. RxSwift's scheduling doesn't work (as far as I can tell), but we can use their observable operators for stuff like this.

Reply:
Is this going to add a lot of effort for new contributors?

Reply:
A good debate that I don't have answers for at the moment. On the surface I'm inclined to say other concurrency primitives are even more difficult to wrap your head around and the declarative nature of Rx and similar libraries make code easier to comprehend.

Reply:
Maybe we should run with something like this and then use RxSwift in the near future. Is there a way we could contribute a linux workaround for RxSwift scheduling?

Reply:
Yeah I definitely think we should just roll with this at the moment.

I think contributing something to fix the Linux stuff (either through documentation or actually fixing it) would be great

Reply:
Okay cool. As I say, as far as I'm aware this is not a safety concern. Data updates are explicitly serialised. Index updates will have a race but that doesn't result in any crash